SU2_CFD/src/output/filewriter/CCGNSFileWriter.cpp

```diff
@@ -0,0 +1,310 @@
+/*!
+ * \file CCGNSFileWriter.cpp
+ * \brief Filewriter class for CGNS format.
+ * \author G. Baldan
+ * \version 7.2.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2021, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../../include/output/filewriter/CCGNSFileWriter.hpp"
+
+const string CCGNSFileWriter::fileExt = ".cgns";
+
+CCGNSFileWriter::CCGNSFileWriter(string valFileName, CParallelDataSorter* valDataSorter, bool isSurf)
+    : CFileWriter(std::move(valFileName), valDataSorter, fileExt) {
+  isSurface = isSurf;
+}
+
+CCGNSFileWriter::~CCGNSFileWriter() {}
+
+void CCGNSFileWriter::Write_Data() {
+#ifdef HAVE_CGNS
+  /*--- Open the CGNS file for writing.  ---*/
+  initializeMeshFile();
+
+  /*--- Write point coordinates. ---*/
+  WriteCoordinate(1, "CoordinateX");
```

Comment:
Is this possible? The code of WriteField and WriteCoordinate looked duplicated.
```suggestion
  WriteField(0, "CoordinateX");
```